File: 3484-design-spreadsheet/3484-design-spreadsheet.cs
public class Spreadsheet {
    private Dictionary<string, int> sheet;

    public Spreadsheet(int rows) {
        sheet = new();
    }
    
    public void SetCell(string cell, int value) {
        sheet[cell] = value;
    }
    
    public void ResetCell(string cell) {
        sheet[cell] = 0;
    }
    
    public int GetValue(string formula) {
        var refs = formula.Substring(1).Split('+');
        return MapToValue(refs[0]) + MapToValue(refs[1]);
    }

    private int MapToValue(string s) {
        if(char.IsDigit(s[0])) return int.Parse(s);
        return sheet.TryGetValue(s, out int value) ? value : 0;
    }
}

/**
 * Your Spreadsheet object will be instantiated and called as such:
 * Spreadsheet obj = new Spreadsheet(rows);
 * obj.SetCell(cell,value);
 * obj.ResetCell(cell);
 * int param_3 = obj.GetValue(formula);
 */

File: 3484-design-spreadsheet/3484-design-spreadsheet.py
class Spreadsheet:

    def __init__(self, rows: int):
        self.sheet = {}

    def setCell(self, cell: str, value: int) -> None:
        self.sheet[cell] = value

    def resetCell(self, cell: str) -> None:
        self.sheet[cell] = 0

    def getValue(self, formula: str) -> int:
        refs = formula[1:].split('+')
        return self.mapToValue(refs[0]) + self.mapToValue(refs[1])
    
    def mapToValue(self, s: str) -> int:
        if s[0].isdigit(): return int(s)
        return self.sheet.get(s, 0)


# Your Spreadsheet object will be instantiated and called as such:
# obj = Spreadsheet(rows)
# obj.setCell(cell,value)
# obj.resetCell(cell)
# param_3 = obj.getValue(formula)